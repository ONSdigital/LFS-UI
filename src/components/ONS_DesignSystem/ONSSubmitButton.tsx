import React, {Component} from 'react';

interface Props {
    label: string,
    id?: string
    primary: boolean,
    small: boolean,
    loading?: boolean,
    onClick?: (...props: any[]) => void
}

export class ONSSubmitButton extends Component <Props, {}> {

    render() {
        let className = "field btn " + (this.props.loading ? "btn--loader is-loading " : "") + (this.props.primary ? "" : "btn--secondary ") + (this.props.small ? "btn--small " : "");
        return (
            <button id={this.props.id} type="submit" disabled={this.props.loading} className={className} onClick={this.props.onClick} data-testid="submit-button">
                <span className="btn__inner">{this.props.label}</span>
            </button>
        );
    }
}
