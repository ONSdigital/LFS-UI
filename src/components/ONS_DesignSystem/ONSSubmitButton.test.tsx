import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSSubmitButton} from "./ONSSubmitButton";
import sinon from "sinon";
import Adapter from "enzyme-adapter-react-16";

describe("ONS Submit Button Test", () => {
    Enzyme.configure({ adapter: new Adapter() })

    const submitProps = {
        label: "Submit",
        primary: true,
        small: true,
    }

    const clickSubmitProps = {
        label: "Submit",
        primary: false,
        small: false,
        onClick: sinon.spy(),
        onChange: undefined
    }

    const loadingButtonProps = {
        label: "Submit",
        primary: true,
        small: false,
        loading: true,
        onClick: sinon.spy(),
    }

    function wrapper(render: any, props:any) {
        return render(
            <ONSSubmitButton    
                label={props.label}
                primary={props.primary}
                small={props.small}
                loading={props.loading}
                onClick={props.onClick}>
            </ONSSubmitButton>
        )
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, Props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, submitProps).exists()).toEqual(true))

    it("should render with the correct label", () => {
        expect(wrapper(mount, submitProps).find("ONSSubmitButton").getElement().props.label).toEqual(submitProps.label);
    })
 
    it('simulates click events', () => {
        wrapper(mount, clickSubmitProps).find('ONSSubmitButton').simulate('click');
        expect(clickSubmitProps.onClick).toHaveProperty('callCount', 1);
    })
 
    it('displays loading button', () => {
        expect(wrapper(mount, loadingButtonProps).find('button').hasClass('btn--loader is-loading')).toEqual(true)
    })
})