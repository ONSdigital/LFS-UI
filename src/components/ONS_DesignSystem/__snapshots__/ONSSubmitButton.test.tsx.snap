// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Submit Button Test matches Snapshot 1`] = `ShallowWrapper {}`;
